pmm/pmm2-api-tests.groovy

```diff
@@ -73,72 +69,63 @@ pipeline {
             }
         }
-        stage('Start staging') {
-            steps {
+        stage('API Tests Setup')
+        {
+            steps{
+                sh '''
+                    sudo curl -L https://github.com/docker/compose/releases/download/1.29.0/docker-compose-`uname -s`-`uname -m` | sudo tee /usr/local/bin/docker-compose > /dev/null
+                    sudo chmod +x /usr/local/bin/docker-compose
+                    sudo ln -sfn /usr/local/bin/docker-compose /usr/bin/docker-compose
+                    sudo docker-compose --version
+                    sudo docker build -t pmm-api-tests .
+                    git clone --single-branch --branch \${GIT_BRANCH_PMM_AGENT} https://github.com/percona/pmm-agent
+                    cd pmm-agent
+                    sudo PMM_SERVER_IMAGE=\${DOCKER_VERSION} MONGO_IMAGE=\${MONGO_IMAGE} MYSQL_IMAGE=\${MYSQL_IMAGE} POSTGRES_IMAGE=\${POSTGRES_IMAGE} docker-compose up -d
+                    cd ../
+                '''
                 script {
-                    if (!env.SERVER_IP) {
-                        runStaging(DOCKER_VERSION, CLIENT_VERSION, '--addclient=ps,1')
-                    }
-                    else
-                    {
-                        env.VM_IP = env.SERVER_IP
-                        env.PMM_URL = "http://admin:admin@${env.VM_IP}"
-                    }
+                    env.VM_IP = "127.0.0.1"
+                    env.PMM_URL = "http://admin:admin@${env.VM_IP}"
                 }
             }
         }
-        stage('Setup Step') {
-            parallel {
-                stage('Setup Docker')
-                {
-                    steps{
-                        sh '''
-                            sudo yum -y install docker
-                            sudo usermod -aG docker ec2-user
-                            sudo service docker start
-                            sudo docker build -t pmm-api-tests .
-                        '''
-                    }
-                }
-                stage('Sanity Check')
-                {
-                    steps {
-                        sh 'timeout 100 bash -c \'while [[ "$(curl -s -o /dev/null -w \'\'%{http_code}\'\' \${PMM_URL}/ping)" != "200" ]]; do sleep 5; done\' || false'
-                    }
-                }
+        stage('Sanity Check')
+        {
+            steps {
+                sh 'timeout 100 bash -c \'while [[ "$(curl -s -o /dev/null -w \'\'%{http_code}\'\' \${PMM_URL}/ping)" != "200" ]]; do sleep 5; done\' || false'
             }
         }
         stage('Run API Test') {
             steps {
                 sh '''
-                    sudo docker run -e PMM_SERVER_URL=\${PMM_URL} -e PMM_RUN_UPDATE_TEST=1 -e PMM_RUN_STT_TESTS=0 --name ${BUILD_TAG} pmm-api-tests
+                    sudo docker run -e PMM_SERVER_URL=\${PMM_URL} -e PMM_RUN_UPDATE_TEST=1 -e PMM_RUN_STT_TESTS=0 --name ${BUILD_TAG} --network host pmm-api-tests
```

Comment:
Is it really needed to use `--network host`? Docker supports interconnection between containers if they are in same network

Reply:
The API tests docker container is not part of the same container-compose running pmm-server and other DB's